Source/Core/Core/AchievementManager.cpp

```diff
@@ -447,14 +447,18 @@ void AchievementManager::CloseGame()
 {
   {
     std::lock_guard lg{m_lock};
-    m_is_game_loaded = false;
-    m_game_id = 0;
-    m_queue.Cancel();
-    m_unlock_map.clear();
-    m_system = nullptr;
-    ActivateDeactivateAchievements();
-    ActivateDeactivateLeaderboards();
-    ActivateDeactivateRichPresence();
+    if (m_is_game_loaded)
+    {
+      m_is_game_loaded = false;
+      ActivateDeactivateAchievements();
+      ActivateDeactivateLeaderboards();
+      ActivateDeactivateRichPresence();
+      m_game_id = 0;
+      m_unlock_map.clear();
+      rc_api_destroy_fetch_game_data_response(&m_game_data);
```

Comment:
Can you add a call to clear out `m_game_data` after this, just so we're sure it's actually empty and the next time it gets initialized behaves correctly? `std::memset(&m_unlock_map, 0, sizeof(m_unlock_map))` should be enough.